src/Extractor/BaseExtractor.php

```diff
@@ -269,109 +263,44 @@ protected function createOutputCsv(string $outputTable): CsvWriter
         return new CsvWriter($this->getOutputFilename($outputTable));
     }
 
-    /**
-     * @param array $table
-     * @return bool|int
-     */
-    protected function createManifest(array $table)
+    protected function createManifest(ExportConfig $export): void
     {
-        $outFilename = $this->getOutputFilename($table['outputTable']) . '.manifest';
+        $metadataSerializer = $this->getManifestMetadataSerializer();
+        $outFilename = $this->getOutputFilename($export->getOutputTable()) . '.manifest';
 
         $manifestData = [
-            'destination' => $table['outputTable'],
-            'incremental' => $table['incremental'],
+            'destination' => $export->getOutputTable(),
+            'incremental' => $export->isIncremental(),
         ];
 
-        if (!empty($table['primaryKey'])) {
-            $manifestData['primary_key'] = $table['primaryKey'];
+        if ($export->hasPrimaryKey()) {
+            $manifestData['primary_key'] = $export->getPrimaryKey();
         }
 
-        $manifestColumns = [];
-
-        if (isset($table['table']) && !is_null($table['table'])) {
-            $tables = $this->getTables([$table['table']]);
-            if (count($tables) > 0) {
-                $tableDetails = $tables[0];
-                $columnMetadata = [];
-                $sanitizedPks = [];
-                $iterColumns = $table['columns'];
-                if (count($iterColumns) === 0) {
-                    $iterColumns = array_map(function ($column) {
-                        return $column['name'];
-                    }, $tableDetails['columns']);
+        if (!$export->hasQuery()) {
+            $table = $this->getMetadataProvider()->getTable($export->getTable());
+            $allTableColumns = $table->getColumns();
+            $columnMetadata = [];
+            $sanitizedPks = [];
+            $exportedColumns = $export->hasColumns() ? $export->getColumns() : $allTableColumns->getNames();
```

Comment:
predtym `$iterColumns` ... je to zoznam stlpcov, ktore sa exportovali ... bud boli stlpce zadane v configu, ... alebo sa exportovali vsetky stlpce a ich zoznam teda ziskame z metadat.